digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala

```diff
@@ -288,6 +293,50 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
     RoadLinkDAO.LinkTypeDao.getAllLinkType(linkIds).groupBy(_._1)
   }
 
+  def getAllPrivateRoadAssociationNames(newTransaction: Boolean = true): Seq[String] = {
+    if(newTransaction) {
+      withDynSession {
+        LinkAttributesDao.getAllExistingDistinctValues(privateRoadAssociationPublicId)
+      }
+    } else
+      LinkAttributesDao.getAllExistingDistinctValues(privateRoadAssociationPublicId)
+  }
+
+  def getPrivateRoadsByAssociationName(roadAssociationName: String, newTransaction: Boolean = true): Seq[(Long, String, String)] = {
+    val linkIds = getValuesByRoadAssociationName(roadAssociationName, privateRoadAssociationPublicId, newTransaction).map(_._2)
+    val roadLinks = getRoadLinksAndComplementaryByLinkIdsFromVVH(linkIds.toSet, newTransaction)
+
+    val municipalityCodes = roadLinks.map(_.municipalityCode).toSet
+    val municipalitiesInfo = getMunicipalitiesInfo(municipalityCodes, newTransaction)
+    val groupedByNameRoadLinks = roadLinks.groupBy(_.municipalityCode).values.map(_.groupBy(_.roadNameIdentifier))
+
+    groupedByNameRoadLinks.flatMap { municipalityRoadLinks =>
+      municipalityRoadLinks.map { roadLinkAssociation =>
+        val roadName = if(roadLinkAssociation._1.getOrElse(" ").trim.isEmpty) roadWithoutName else roadLinkAssociation._1.get
+        val maxLengthRoadLink = roadLinkAssociation._2.maxBy(_.length)
+       (maxLengthRoadLink.linkId, roadName, municipalitiesInfo.find(_.id == maxLengthRoadLink.municipalityCode).head.name)
+      }
+    }.toSeq
+  }
+
+  def getMunicipalitiesInfo(municipalityCodes: Set[Int], newTransaction: Boolean = true) = {
+    if(newTransaction)
+      withDynSession {
+        municipalityDao.getMunicipalitiesNameAndIdByCode(municipalityCodes)
+      }
+    else
+      municipalityDao.getMunicipalitiesNameAndIdByCode(municipalityCodes)
+  }
```

Comment:
already exist that method on a MunicipalityService **getMunicipalitiesNameAndIdByCode,** please delete that method and use the service